src/main/kotlin/org/wycliffeassociates/otter/common/domain/ImportResourceContainer.kt

```diff
@@ -56,12 +55,11 @@ class ImportResourceContainer(
         val dc = rc.manifest.dublinCore
 
         return Completable.fromCallable {
-            languageDao.getBySlug(dc.language.identifier).subscribe {
-                val resourceMetadata = dc.mapToMetadata(container, it)
-                //set the id in the resourceMetadata object once it returns from the insert call
+            languageRepository.getBySlug(dc.language.identifier).subscribe { language ->
+                val resourceMetadata = dc.mapToMetadata(container, language)
                 //metadata id is going to be needed for the collection insert
-                metadataDao.insert(resourceMetadata).subscribe {
-                    resourceMetadata.id = it
```

Comment:
I believe this line is still needed; import project needs the id field to be set